pkgs/tools/networking/p2p/tahoe-lafs/default.nix

```diff
@@ -50,7 +50,7 @@ pythonPackages.buildPythonApplication rec {
 
   nativeBuildInputs = with pythonPackages; [ sphinx texinfo ];
 
-  buildInputs = with pythonPackages; [ unzip numpy mock ];
+  buildInputs = with pythonPackages; [ unzip mock wheel ];
```

Comment:
`unzip` doesn't seem to be necessary (anymore?), and I'd put `mock` into `checkInputs`.

`numpy` still seems to be used in a separate component:

> - The provisioning/reliability pages were removed from the main client's web
>   interface, and moved into a standalone web-based tool. Use the > ``run.py``
>   script in ``misc/operations_helpers/provisioning/`` to access them.
> - A new "reliability" page was added, with the beginnings of work on a
>  statistical loss model. You can tell this page how many servers you 
>  are using and their independent failure probabilities, and it will 
>  tell you the likelihood that an arbitrary file will survive each  
>  repair period. The "numpy" package must be installed to access this
>  page. A partial paper, written by Shawn Willden, has been added to 
>  docs/proposed/lossmodel.lyx .

Do we ship this component?


`wheel` seems to only be used in `misc/coding_tools/graph-deps.py`, which looks like a script to graph dependencies while developing on tahoe-lafs itself - I guess we don't need it either, at least I could run the tests without it, didn't try to run it myself :-)

Reply:
> wheel seems to only be used in misc/coding_tools/graph-deps.py, which looks like a script to graph dependencies while developing on tahoe-lafs itself - I guess we don't need it either, at least I could run the tests without it, didn't try to run it myself :-)

I found that the build fails without the wheel dependency.  I don't claim to understand why.  I guess some part of the nix python toolchain is trying to build a wheel.  Maybe only on aarch64?  I put part of the error message into the commit message.  I can reproduce it completely if that would be helpful.

> The provisioning/reliability pages were removed from the main client's web interface, and moved into a standalone web-based tool. Use the > run.py script in misc/operations_helpers/provisioning/ to access them.

Bleh, I looked all over for a numpy dependency but I guess not really _all_ over...  I've never used this tool or heard of anyone using this tool.  I don't see any evidence in default.nix that anything in misc is included in the output.  So ... I think it doesn't matter?  But I don't have a working install at the moment to verify against.

Reply:
Here's my result building on aarch64 without an explicit wheel buildInput:

```
building '/nix/store/cn5ly43g6cp0dbz2jfwwb14yjd9fhbm0-tahoe-lafs-1.13.0.drv'...
unpacking sources
unpacking source archive /nix/store/x7ly97zfhsdi9yjf9a12fpaml5jz1rvb-tahoe-lafs-1.13.0.tar.bz2
source root is tahoe-lafs-1.13.0
setting SOURCE_DATE_EPOCH to timestamp 1533495968 of file tahoe-lafs-1.13.0/setup.cfg
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
building
usage: nix_run_setup [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: nix_run_setup --help [cmd1 cmd2 ...]
   or: nix_run_setup --help-commands
   or: nix_run_setup cmd --help

error: invalid command 'bdist_wheel'
builder for '/nix/store/cn5ly43g6cp0dbz2jfwwb14yjd9fhbm0-tahoe-lafs-1.13.0.drv' failed with exit code 1
cannot build derivation '/nix/store/4fprb64l5q82v4s183bgc6n4r0fxsxhs-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/kkl6i9z3y4kzan60j2jbakl6zwswpx0b-nixos-system-nixos-19.09pre174594.0c0954781e2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/189kynzjvgqrlv2q723yn3np939k1g88-nixos-sd-image-19.09pre174594.0c0954781e2-aarch64-linux.img.drv': 1 dependencies couldn't be built
error: build of '/nix/store/189kynzjvgqrlv2q723yn3np939k1g88-nixos-sd-image-19.09pre174594.0c0954781e2-aarch64-linux.img.drv' failed
```

Reply:
These three dependencies should be moved to `nativeBuildInputs`, `propagatedBuildInputs`, or `checkInputs`. See e.g. https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters.